common/src/chain/block/block_reward.rs

```diff
@@ -88,7 +88,11 @@ impl<'a> Signable for BlockRewardTransactable<'a> {
 }
 
 impl<'a> Transactable for BlockRewardTransactable<'a> {
-    fn signatures(&self) -> Option<&[InputWitness]> {
-        self.witness
+    fn signature(&self, index: usize) -> Option<&InputWitness> {
+        self.witness.and_then(|w| w.get(index))
+    }
+
+    fn signatures(&self) -> Vec<Option<InputWitness>> {
+        self.witness.map_or(vec![], |w| w.iter().map(|s| Some(s.clone())).collect())
     }
```

Comment:
Why is this necessary? You can manipulate the slice in any way you like outside of here. This should be minimal.

Reply:
I want `verify_signature` (which requires `Transactable` trait) to work with `PartiallySignedTransaction` so that it can be used in the wallet.
The problem is that I cannot implement single getter for `Transaction` which stores `Vec<Witness>` and `PartiallySignedTransaction` which stores `Vec<Option<Witness>>` without copying data. As a partial optimisation I added new method `signature(index)` to get a reference without collect call.

Reply:
Copying data for PartiallySignedTransaction is fine. It's meant to be serialized anyway.

Reply:
Please revert this before merging.